components/ChatMenu.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <component
+    :is="chatType"
+    :id="smallScreen ? 'menu-option-chat-sm' : 'menu-option-chat'"
+    :class="smallScreen ? 'text-white mr-3 position-relative' : 'text-center small p-0'"
+    href="#"
+    aria-label="chats"
+    @click="toChats"
+  >
+    <div class="position-relative">
+      <v-icon name="comments" scale="2" class="chat__icon" />
+      <div v-if="!smallScreen" class="nav-item__text">
+        Chats
+      </div>
+      <b-badge v-if="chatCount" variant="danger" class="chatbadge">
+        {{ chatCount }}
+      </b-badge>
+    </div>
+  </component>
+</template>
+
+<script>
+export default {
+  name: 'ChatMenu',
+  props: {
+    smallScreen: {
+      type: Boolean,
+      required: false,
+      default: false
+    }
+  },
+  computed: {
+    chatType() {
+      return this.smallScreen ? 'a' : 'b-nav-item'
+    },
+    chatCount() {
+      // Don't show so many that the layout breaks.
+      return Math.min(99, this.$store.getters['chats/unseenCount'])
+    }
+  },
+  watch: {
+    chatCount: function() {
+      this.$emit('chat-count', this.chatCount)
```

Comment:
Wouldn't it be better to emit update:chat-count so that the parent can use .sync?

Reply:
I avoided using sync as it's been deprecated in v3 hasn't it?